Reduce unsafeness in NetworkProcess and remove redundant protected methods

annevk opened the pull request:
#### da452c6c8bc7920e913a2a3e9a067f33439f9c17
<pre>
Reduce unsafeness in NetworkProcess and remove redundant protected methods
<a href="https://bugs.webkit.org/show_bug.cgi?id=295623">https://bugs.webkit.org/show_bug.cgi?id=295623</a>

Reviewed by Youenn Fablet.

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a> by
addressing a variety of static analyzer warnings in NetworkProcess code
and increasing the number of smart pointer member variables that are
marked const. For those that have a trivial getter, we also remove the
protected method, if any.

Canonical link: <a href="https://commits.webkit.org/297167@main">https://commits.webkit.org/297167@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/141e36c668313d70dc877cb26004bd82e6357da3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110739 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30402 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20834 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116766 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/61007 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31082 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38988 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/84208 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113687 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24824 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99721 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64649 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24194 "Passed tests") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/60561 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94220 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17915 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119556 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37784 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/28082 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/93170 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38160 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95986 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92994 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38025 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15773 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33758 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17865 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37677 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/43147 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37338 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40676 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/39045 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jelee53 commented on Source/WebKit/NetworkProcess/storage/LocalStorageManager.h:
```diff
@@ -76,7 +76,7 @@ class LocalStorageManager {
 
     String m_path;
-    CheckedRef<StorageAreaRegistry> m_registry;
+    const CheckedRef<StorageAreaRegistry> m_registry;
```
Hi @annevk  - I think we accidentally did the same work 😅. 

There's a decent amount of overlap with this PR and this one: https://github.com/WebKit/WebKit/pull/47750. I think we have four overlapping files! How do you think we should handle this? 

(Actually maybe I'll put this in Slack but tomorrow because it's 1AM here.)

annevk replied:
Oh I'm sorry. From a brief skim it looks like a lot of it is complementary fortunately.

youennf commented on Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
```diff
@@ -1697,7 +1697,7 @@ void NetworkConnectionToWebProcess::navigatorSubscribeToPushService(URL&& scopeU
 
     auto registrableDomain = RegistrableDomain(scopeURL);
-    session->protectedNotificationManager()->subscribeToPushService(WTFMove(scopeURL), WTFMove(applicationServerKey), [weakThis = WeakPtr { *this }, completionHandler = WTFMove(completionHandler), registrableDomain = WTFMove(registrableDomain)] (Expected<PushSubscriptionData, ExceptionData>&& result) mutable {
+    session->notificationManager().subscribeToPushService(WTFMove(scopeURL), WTFMove(applicationServerKey), [weakThis = WeakPtr { *this }, completionHandler = WTFMove(completionHandler), registrableDomain = WTFMove(registrableDomain)] (Expected<PushSubscriptionData, ExceptionData>&& result) mutable {
```
s/Expected<PushSubscriptionData, ExceptionData>/auto/ ?

youennf commented on Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerConnection.cpp:
```diff
@@ -782,6 +786,11 @@ NetworkSession* WebSWServerConnection::session()
     return protectedNetworkProcess()->networkSession(sessionID());
 }
 
+CheckedPtr<NetworkSession> WebSWServerConnection::checkedSession()
```
Can it be inline?

annevk replied:
Not without including `NetworkSession.h` which I suspect we don't want to do.

annevk committed:
Reduce unsafeness in NetworkProcess and remove redundant protected methods
https://bugs.webkit.org/show_bug.cgi?id=295623

Reviewed by Youenn Fablet.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines by
addressing a variety of static analyzer warnings in NetworkProcess code
and increasing the number of smart pointer member variables that are
marked const. For those that have a trivial getter, we also remove the
protected method, if any.

Canonical link: https://commits.webkit.org/297167@main
